### PR TITLE
improve the performance of dbs file insert

### LINKED ISSUE #8740: WorkQueueManagerWMBSFileFeeder slowly adding data to dbsbuffer tables
Related to https://github.com/dmwm/WMCore/issues/8528

I've seen WorkQueueManagerWMBSFileFeeder heartbeat alarms since the past weekend, which is hitting us in at least a couple of agents (vocms0252, vocms0254).

Checking the logs on vocms0252...
```
2018-08-12 15:19:52,946:140043577100032:INFO:WorkQueue:Adding WMBS subscription for vlimant_ACDC0_task_HIG-RunIIFall17wmLHEGS-01118__v1_T_180808_124312_4167
... almost 2 days later ...
2018-08-14 13:11:00,644:140043577100032:INFO:WMBSHelper:Inserting in bulk all the file info into WMBS for vlimant_ACDC0_task_HIG-RunIIFall17wmLHEGS-01118__v1_T_180808_124312_4167
2018-08-14 13:25:55,078:140043577100032:INFO:WMBSHelper:Inserting all the file info into DBSBuffer for vlimant_ACDC0_task_HIG-RunIIFall17wmLHEGS-01118__v1_T_180808_124312_4167
... and now slowly going through these dbsbuffer insertions...
```

and vocms0254 seems to be stuck processing this ACDC
```
2018-08-13 16:30:33,521:139965766612736:INFO:WorkQueue:Adding WMBS subscription for vlimant_ACDC0_task_HIG-RunIIFall17wmLHEGS-01122__v1_T_180808_130708_5376
```

### PR DESCRIPTION
**Pull request by ticoann**
fixes #8740

### CODE REVIEW

**Comment by ticoann** [comment]
Not tested. One other question I have is whether we need to add unmerged ACDC files in DBSBuffer.
I guess if we have parents files we need to add it to tract it but in case we don't have parents we might not need this. (In that case, I can create another patch to skip that)

**Review by amaltaro — APPROVED** [review]
Patch looks good, Seangchan. I'm adding it to my testbed node to test with all the other patches that we'll have to backport.

**Comment by ticoann** [comment]
@amaltaro, Alan I added code to skip the unmerged acdc files without parents for inserting dbsbuffer.

File: src/python/WMCore/WorkQueue/WMBSHelper.py [hunk 1]
```diff
@@ -710,8 +710,11 @@ def _addACDCFileToWMBSFile(self, acdcFile, inFileset=True):
             wmbsFile.addRun(run)
 
         dbsFile = self._convertACDCFileToDBSFile(acdcFile)
-        self._addToDBSBuffer(dbsFile, checksums, acdcFile["locations"])
 
+        if "/store/unmerged/" not in dbsFile["LogicalFileName"] or dbsFile["ParentList"]:
```
Comment (amaltaro) [inline src/python/WMCore/WorkQueue/WMBSHelper.py]: shouldn't it be `if not dbsFile["LogicalFileName"].startswith("/store/unmerged/")...`

Comment (ticoann) [reply src/python/WMCore/WorkQueue/WMBSHelper.py]: I can changed it but is there case if /store/unmerged/ is in the string but actually not a unmerged file?

**Comment by amaltaro** [comment]
You're right Seangchan. I mixed things up and I thought that comparison wouldn't work, sorry!

**Comment by ticoann** [comment]
> You're right Seangchan. I mixed things up and I thought that comparison wouldn't work, sorry!

I thought you just want to improve the code. :smile: I updated the patch anyway.

**Comment by amaltaro** [comment]
Seangchan, I didn't test it after you added the /store/unmerged handle, but it should be fine to go in for further tests.

**Comment by amaltaro** [comment]
and we might have to push this fix in to vocms0254, which is struggling with an ACDC workflow (almost 200k files):
vlimant_ACDC0_task_SUS-RunIIFall17wmLHEGS-00016__v1_T_180807_032626_4371

**Pull request merged**